api/envoy/extensions/clusters/dynamic_forward_proxy/v3/cluster.proto

```diff
@@ -53,4 +56,21 @@ message ClusterConfig {
   //   <https://datatracker.ietf.org/doc/html/rfc7540#section-9.1.1>`_
   //
   bool allow_coalesced_connections = 3;
+
+  // Enable strict dns sub cluster instead logic dns.
```

Comment:
Can you flesh this out. From an end user perspective it's hard to understand what this means and why I would use it.